### ISSUE TITLE
Is it possible to remove the deprecated curry methods on Function[0..n]?

### ISSUE THREAD

**Issue by johnmcclean-aol** [jOOQ/jOOL#274]
I can submit a PR if you like. 

I think one curry() method per Function type that returns a chain of single parameter functions would be useful here instead.

**Comment by lukaseder** [comment]
>I can submit a PR if you like.

Glad you asked. Don't :) Because this code is generated. The real code (generator code) is not in this repository.

The deprecation was introduced with https://github.com/jOOQ/jOOL/issues/216, because the functions don't actually curry, they apply partially.

Do you need curry at your side?

**Comment by johnmcclean-aol** [comment]
Yeah, I'd like a curry method on them directly. I'm going extend them (to add in useful methods related to the types in cyclops), a curry method would be very useful (could be added in either library) - but the deprecated curry methods will be confusing in that case.

By the way - what do you think of this issue - https://github.com/aol/cyclops-react/issues/422 . Do you think we could add such a match method directly to the Tuples in jOOλ?

**Comment by lukaseder** [comment]
OK, thanks for the feedback. I guess we can remove them, as jOOλ hasn't reached 1.0 yet.

Hmm, I'd prefer not to put too many things in the `Tuple[N]` types prematurely. For instance, your suggested `match()` method lists [N] predicates per tuple. But are you sure those predicates will always operate on only one attribute?

**Comment by lukaseder** [comment]
Anyway, I think we should discuss a potential new `match()` method in a new issue...

**Comment by lukaseder** [comment]
It might still be a bit early to remove the methods. I'll postpone this to the next release.
